File: movie_tickets.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

int main() {
    unordered_map<string, vector<string>> movies;
    int Q;
    cin >> Q;

    while (Q--) {
        string op;
        cin >> op;

        if (op == "BOOK") {
            string x, y;
            cin >> x >> y;

            if (find(movies[y].begin(), movies[y].end(), x) != movies[y].end() 
                || movies[y].size() >= 100) {
                cout << "false\n";
            } else {
                movies[y].push_back(x);
                cout << "true\n";
            }
        }

        else if (op == "CANCEL") {
            string x, y;
            cin >> x >> y;

            if (find(movies[y].begin(), movies[y].end(), x) != movies[y].end()) {
                movies[y].erase(find(movies[y].begin(), movies[y].end(), x));
                cout << "true\n";
            } else {
                cout << "false\n";
            }
        }

        else if (op == "IS_BOOKED") {
            string x, y;
            cin >> x >> y;

            if (find(movies[y].begin(), movies[y].end(), x) != movies[y].end()) {
                cout << "true\n";
            } else {
                cout << "false\n";
            }
        }

        else if (op == "AVAILABLE_TICKETS") {
            string y;
            cin >> y;

            cout << 100 - movies[y].size() << "\n";
        }
    }
}

File: movie_tickets.py
movies = {}  # movie_id -> list of users

Q = int(input())

for _ in range(Q):
    q = input().split()

    if q[0] == "BOOK":
        x, y = q[1], q[2]

        if y not in movies:
            movies[y] = []

        if x in movies[y] or len(movies[y]) >= 100:
            print("false")
        else:
            movies[y].append(x)
            print("true")

    elif q[0] == "CANCEL":
        x, y = q[1], q[2]

        if y in movies and x in movies[y]:
            movies[y].remove(x)
            print("true")
        else:
            print("false")

    elif q[0] == "IS_BOOKED":
        x, y = q[1], q[2]

        if y in movies and x in movies[y]:
            print("true")
        else:
            print("false")

    elif q[0] == "AVAILABLE_TICKETS":
        y = q[1]

        if y in movies:
            print(100 - len(movies[y]))
        else:
            print(100)